aea/crypto/fetchai.py

```diff
@@ -222,10 +222,16 @@ def validate_transaction(
 
         :return: True if the random_message is equals to tx['input']
         """
+        import pdb
+
+        pdb.set_trace()
         tx_contents = cast(TxContents, self._api.tx.contents(tx_digest))
-        is_valid = tx_contents.from_address.to_hex() == client
+        transfers = tx_contents.transfers
+        transfers_result = False
+        for k in transfers.keys():
```

Comment:
You have the address, and the key is the address. So just check that the value for the key matches the amount.